Everserver tests fail after logging change

Following merge of https://github.com/equinor/ert/pull/11158 CI tests has started failing:

```
[gw1] [ 99%] FAILED tests/everest/test_everserver.py::test_status_exception@starts_everest 
tests/everest/test_everserver.py::test_status_max_batch_num@starts_everest 
[gw1] [ 99%] FAILED tests/everest/test_everserver.py::test_status_max_batch_num@starts_everest 
tests/everest/test_everserver.py::test_status_contains_max_runtime_failure@starts_everest 
[gw1] [ 99%] FAILED tests/everest/test_everserver.py::test_status_contains_max_runtime_failure@starts_everest 
tests/everest/test_logging.py::test_logging_setup@starts_everest 
[gw1] [100%] PASSED tests/everest/test_logging.py::test_logging_setup@starts_everest 

=================================== FAILURES ===================================
_____________________________ test_https_requests ______________________________
[gw1] linux -- Python 3.11.12 /home/runner/work/ert/ert/.venv/bin/python

copy_math_func_test_data_to_tmp = None

    @pytest.mark.integration_test
    @pytest.mark.skip_mac_ci
    @pytest.mark.xdist_group(name="starts_everest")
    async def test_https_requests(copy_math_func_test_data_to_tmp):
        everest_config = EverestConfig.load_file("config_minimal.yml")
        everest_config.forward_model.append(ForwardModelStepConfig(job="sleep 5"))
        everest_config.install_jobs.append(
            InstallJobConfig(name="sleep", executable=f"{which('sleep')}")
        )
        # start_server() loads config based on config_path, so we need to actually
        # overwrite it
        everest_config.dump("config_minimal.yml")
    
        status_path = ServerConfig.get_everserver_status_path(everest_config.output_dir)
        expected_server_status = ExperimentState.never_run
        assert expected_server_status == everserver_status(status_path)["status"]
        makedirs_if_needed(everest_config.output_dir, roll_if_exists=True)
        await start_server(everest_config, logging_level=logging.INFO)
    
>       wait_for_server(everest_config.output_dir, 240)

/home/runner/work/ert/ert/tests/everest/test_detached.py:61: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

output_dir = '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_https_requests0/everest_output'
timeout = 240

    def wait_for_server(output_dir: str, timeout: int | float) -> None:
        """
        Checks everest server has started _HTTP_REQUEST_RETRY times. Waits
        progressively longer between each check.
    
        Raise an exception when the timeout is reached.
        """
        sleep_time_increment = float(timeout) / (2**_HTTP_REQUEST_RETRY - 1)
        for retry_count in range(_HTTP_REQUEST_RETRY):
            if server_is_running(*ServerConfig.get_server_context(output_dir)):
                return
            else:
                time.sleep(sleep_time_increment * (2**retry_count))
>       raise RuntimeError("Failed to get reply from server within configured timeout.")
E       RuntimeError: Failed to get reply from server within configured timeout.

/home/runner/work/ert/ert/src/everest/detached/client.py:145: RuntimeError
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/logging/config.py", line 573, in configure
    handler = self.configure_handler(handlers[name])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/logging/config.py", line 757, in configure_handler
    result = factory(**kwargs)
             ^^^^^^^^^^^^^^^^^
  File "/home/runner/work/ert/ert/src/ert/logging/__init__.py", line 51, in __init__
    super().__init__(filename, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/logging/__init__.py", line 1181, in __init__
    StreamHandler.__init__(self, self._open())
                                 ^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/logging/__init__.py", line 1213, in _open
    return open_func(self.baseFilename, self.mode,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pytest-of-runner/pytest-0/popen-gw1/test_https_requests0/logs/api-log-storage-2025-06-20T06:59.txt'
```

--- a/src/ert/logging/__init__.py
+++ b/src/ert/logging/__init__.py
@@ -4,6 +4,7 @@ import pathlib
 import sys
 from collections.abc import Callable
 from datetime import datetime
+from pathlib import Path
 from types import TracebackType
 from typing import Any
 
@@ -47,6 +48,7 @@ class TimestampedFileHandler(logging.FileHandler):
         if kwargs.pop("use_log_dir_from_env", False) and "ERT_LOG_DIR" in os.environ:
             log_dir = os.environ["ERT_LOG_DIR"]
             filename = log_dir + "/" + filename
+            Path(filename).parent.mkdir(exist_ok=True, parents=True)
 
         super().__init__(filename, *args, **kwargs)
 

